[IMP] helpdesk_fieldservice (Close Ticket Wizard)

patrickrwilson opened the pull request:
Adds a close ticket wizard that is triggered when the fso is being completed and the related ticket is still open.

patrickrwilson committed:
[IMP] Close Ticket Wizard

Adds a close ticket wizard that is triggered when the fso is being completed and the related ticket is still open.

max3903 commented on helpdesk_fieldservice/wizard/fsm_order_close_wizard.py:
```diff
@@ -0,0 +1,60 @@
+    @api.multi
+    def action_close_ticket(self):
+        for record in self:
+            if record.ticket_id.stage_id.is_close is False:
```
```suggestion
            if not record.ticket_id.stage_id.is_close:
```

max3903 replied:
@patrickrwilson What if the ticket has multiple FSM orders and one of them is not closed?

patrickrwilson replied:
good point, i'll include that logic

max3903 commented on helpdesk_fieldservice/wizard/fsm_order_close_wizard.py:
```diff
@@ -0,0 +1,60 @@
+            'tag': 'reload',
+        }
+
+class FSMOrder(models.Model):
```
@patrickrwilson This class can go in `models/fsm_order.py`.

max3903 commented on helpdesk_fieldservice/wizard/fsm_order_close_wizard.xml:
```diff
@@ -0,0 +1,39 @@
+                         required="1"/>
+              </group>
+              <footer>
+                  <button string="Update Ticket"
```
```suggestion
                  <button string="Close Ticket"
```

max3903 commented on helpdesk_fieldservice/wizard/fsm_order_close_wizard.py:
```diff
@@ -0,0 +1,60 @@
+# Copyright (C) 2019 Pavlov Media
+# License Proprietary. Do not copy, share nor distribute.
```
```suggestion
# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```

max3903 commented on helpdesk_fieldservice/wizard/fsm_order_close_wizard.xml:
```diff
@@ -0,0 +1,39 @@
+<!-- Copyright (C) 2019 Pavlov Media
+    License Proprietary. Do not copy, share nor distribute.-->
```
```suggestion
    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
```

max3903 commented on helpdesk_fieldservice/wizard/__init__.py:
```diff
@@ -0,0 +1,4 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators
```
```suggestion
# Copyright (C) 2019 Pavlov Media
```

patrickrwilson committed:
[UPD] Reviewer Suggestions

Reviewer suggestions.

osi-scampbell reviewed (changes requested):
In helpdesk_fieldservice/models/fsm_order.py we need to import ValidationError and _ or else it will throw an error, also remove those objects from the import list on fsm_order_close_wizard.py because we don't use them on that file

max3903 commented:
@patrickrwilson Can you fix the Travis errors?
https://travis-ci.org/ursais/osi-addons/jobs/611966871#L327

Thanks!

osi-scampbell commented:
> @patrickrwilson Can you fix the Travis errors?
> https://travis-ci.org/ursais/osi-addons/jobs/611966871#L327
> 
> Thanks!

@patrickrwilson If you import those two objects I mentioned in my review it will take care of everything except helpdesk_fieldservice/models/fsm_order.py line 32. Looks like that line is 1 character too long. Then the travis should be green!

patrickrwilson committed:
[IMP] Added check for multiple fso on ticket

Added the check if there are other open fso's on the ticket then it won't prompt to close the ticket.

patrickrwilson committed:
[FIX] Travis and imports

Fixed travis errors and missing imports

patrickrwilson commented:
@max3903 @osi-scampbell Thanks for the reviews, i just sent the last commit which i think has all the fixes that you brought up.

max3903 committed:
Update fsm_location.xml
